lib/livebook/hubs/team_client.ex

```diff
@@ -244,9 +244,19 @@ defmodule Livebook.Hubs.TeamClient do
     }
   end
 
-  defp build_deployment_group(state, %{id: id, name: name, mode: mode, secrets: secrets}) do
-    secrets = Enum.map(secrets, &build_secret(state, &1))
-    %DeploymentGroup{id: id, name: name, mode: mode, hub_id: state.hub.id, secrets: secrets}
+  defp build_deployment_group(state, deployment_group) do
+    secrets = Enum.map(deployment_group.secrets, &build_secret(state, &1))
+
+    %DeploymentGroup{
+      id: deployment_group.id,
+      name: deployment_group.name,
+      mode: deployment_group.mode,
+      hub_id: state.hub.id,
+      secrets: secrets,
+      clustering: deployment_group.clustering,
+      zta_provider: String.to_atom(deployment_group.zta_provider),
+      zta_key: deployment_group.zta_key
```

Comment:
We need to encrypt the value, same as secrets value.

Reply:
We don't need to encrypt the zta_key because as far as I know it doesn't keep any confidential/secret information.

Reply:
I see, but even then, do we want to assume that for any future providers we may add?

Reply:
Then we can encrypt it for certain providers before sending to the server or something. Glad to revisit later. :)